### PR TITLE
Allow published_at to be within 15 minutes, not 15 hours

### PR DESCRIPTION
**Pull request by lightalloy**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [x] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description
We only allow future or current published_at for the new articles, but we allow a 15-minute gap, just in case and for convinience when using editor v1 and the user chooses to specify `published_at`.
This pr allows `published_at` to be within 15 minutes in the past (as intended), not 15 hours

## Related Tickets & Documents

- Related Issue #15858 

## QA Instructions, Screenshots, Recordings
- enable the feature flag `FeatureFlag.enable(:schedule_articles)`
You can use both editors for the next steps:
- try create an article with `published_at` set to 1 hour ago, press `save` - an article should't be saved, an error message should be displayed
- try create an article with `published_at` set to 10 minutes ago, press `save` - an article should be saved

## Added/updated tests?

- [x] Yes
- [ ] No, and this is why: _please replace this line with details on why tests
      have not been included_
- [ ] I need help with writing tests

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

- [ ] I've updated the [Developer Docs](https://developers.forem.com) or
      [Storybook](https://storybook.forem.com/) (for Crayons components)
- [ ] This PR changes the Forem platform and our documentation needs to be
      updated. I have filled out the
      [Changes Requested](https://github.com/forem/admin-docs/issues/new?assignees=&labels=&template=changes_requested.md)
      issue template so Community Success can help update the Admin Docs
      appropriately.
- [ ] I've updated the README or added inline documentation
- [ ] I've added an entry to
      [`CHANGELOG.md`](https://github.com/forem/forem/tree/main/CHANGELOG.md)
- [ ] I will share this change in a [Changelog](https://forem.dev/t/changelog)
      or in a [forem.dev](http://forem.dev) post
- [ ] I will share this change internally with the appropriate teams
- [ ] I'm not sure how best to communicate this change and need help
- [x] This change does not need to be communicated, and this is why not: a small bug fix

### CODE REVIEW

**Review by Ridhwana — APPROVED** [review]

File: app/models/article.rb [hunk 1]
```diff
@@ -817,7 +817,7 @@ def validate_co_authors_exist
 
   def future_or_current_published_at
     # allow published_at in the future or within 15 minutes in the past
-    return if !published || published_at > Time.current - (15 * 60 * 60)
+    return if !published || published_at > Time.current - (15 * 60)
```
Comment (fdocr) [inline app/models/article.rb:820]: **Suggestion:** Maybe we can use ActiveSupport's helper methods to make it more readable and less error prone 🙂 

```suggestion
    return if !published || published_at > 15.minutes.ago
```

Comment (lightalloy) [reply app/models/article.rb]: I don't know why I haven't used that :D

**Review by jaw6 — APPROVED** [review]
I suspect this will need some adjustment to account for https://github.com/forem/forem/pull/18353, but the PR as is looks good to me :+1:

**Comment by lightalloy** [comment]
@jaw6 this validation only run on create, so it shouldn't be affected by making `published_at` immutable for already published articles.

**Pull request merged**